graphql-dgs/src/main/kotlin/com/netflix/graphql/dgs/federation/DefaultDgsFederationResolver.kt

```diff
@@ -49,29 +46,39 @@ open class DefaultDgsFederationResolver() : DgsFederationResolver {
 
     private fun dgsEntityFetchers(env: DataFetchingEnvironment): CompletableFuture<List<Any?>> {
         val resultList = env.getArgument<List<Map<String, Any>>>(_Entity.argumentName)
-                .stream()
-                .map { values ->
-                    val fetcher = dgsSchemaProvider.entityFetchers[values["__typename"]]
+            .stream()
+            .map { values ->
+                val typename = values["__typename"] ?: throw RuntimeException("__typename missing from arguments in federated query")
+                val fetcher = dgsSchemaProvider.entityFetchers[typename]
```

Comment:
This is a good fix. We've had some support questions around this because entity fetcher was missing.